examples/nodejs/neo/sync.js

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env node
+
+/**
+ * NeoBlockchain Syncing Example
+ *
+ * An example usage to have a running NeoBlockchain instance syncing in the background while 
+ * perform user tasks at foreground. Please make sure there is a local MongoDB server running.
+ *
+ * To verify stored data, this example will interact with MongoDB collections directly using 
+ * mongoose library.
+ * 
+ * WARNING: This demo will result in reset of data of specified database.
+ */
+
+// -- Bootstrap
+
+const Neo = require('../../../dist/neo.blockchain.neo').neo
+const mongoose = require('mongoose')
+
+// -- Chain of command
+
+async function main () {
+  console.log('== NeoBlockchain Syncing Example ==')
+
+  // Instantiate a 'full' mode Neo blockchain against 'testnet' network
+  const neoBlockchain = new Neo('full', 'testnet')
+
+  // Verify and establish MongoDB connection with hard coded connection string
+  try {
+    await mongoose.connect('mongodb://localhost/sync_demo', { useMongoClient: true })
+  } catch (err) {
+    console.error('Failed to establish MongoDB connection. err:', err)
+    console.log('Exiting...')
+    process.exit()
+  }
+
+  // Drop all existing collections in database in order to start fresh
+  console.log('existing collections:')
+  const collectionsInfo = await mongoose.connection.db.listCollections().toArray()
+  collectionsInfo.forEach(async (info) => {
+    console.log(`> dropping [${info.name}] collection...`)
+    await mongoose.connection.db.dropCollection(info.name)
+  })
+
+  // Allow it to sync for 30 seconds
+  console.log('Start syncing for 30 seconds...')
+  neoBlockchain.sync.start()
+  await sleep(30000)
+  neoBlockchain.sync.stop()
+  console.log('Stop syncing')
+
+  // Refer to already defined models (defined in 'mongodb.js')
+  const BlockModel = mongoose.model('b_neo_t_blocks')
+  const AddressModel = mongoose.model('b_neo_t_addresses')
+  const TransactionModel = mongoose.model('b_neo_t_transactions')
+
+  // Report document counts per collection type
+  console.log('block count:', await BlockModel.count({}))
+  console.log('address count:', await AddressModel.count({}))
+  console.log('transaction count:', await TransactionModel.count({}))
+
+  // Cherry pick a block, address and transaction document
+  const cherryBlock = await BlockModel.findOne({ index: 5953 })
```

Comment:
The block, txid, and address may not exist if the user is unable to sync 6000 blocks in 30s due to bandwidth limitations.  consider adding a comment about this in the header comment.